common/os/Mutex.h

```diff
@@ -38,7 +38,7 @@ namespace os {
 
   class AutoMutex {
   public:
-    AutoMutex(Mutex* mutex) { m = mutex; m->lock(); }
+    AutoMutex(Mutex* mutex, bool already_locked = false) { m = mutex; if (!already_locked) m->lock(); }
```

Comment:
I didn't see anywhere calling with `already_locked=true`. Is this change necessary?